starlette/responses.py

```diff
@@ -148,6 +149,12 @@ def delete_cookie(
         )
 
     async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
+        if (
+            self.background is not None
+            and background.is_background_task_middleware_installed(scope)
+        ):
+            background.add_tasks(scope, self.background)
+            self.background = None
```

Comment:
What's the scenario you are trying to prevent when setting background to `None`? What if a user has a global response object?

```py
response = Response(background=BackgroundTasks(...))
```

Reply:
Ah, I see now. You are doing because you didn't remove the logic below where we run the background task.

I think we can remove that logic, and you can avoid this line.

(Is my understanding correct?)